### PR TITLE
Error component

### PR DESCRIPTION
**Pull request by summer-cook**
# summary
- creates an error component to be used on pages that might throw an error

# related
https://github.com/scientist-softserv/webstore-component-library/issues/25

# screenshot
default (this example has multiple errors so would be in dev)
image

alternate (dismissible)
image

### CODE REVIEW

File: src/components/Error/Error.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { Alert, Button, Container } from 'react-bootstrap'
+
+const Error = ({ errorText, errorTitle, router, variant }) => {
+  return (
+    <Container>
+      <Alert className='mt-5 text-break' variant={variant}>
+        {errorTitle && (
+          <Alert.Heading>{errorTitle}</Alert.Heading>
+        )}
+        {errorText && (errorText)}
```
Comment (alishaevn) [inline src/components/Error/Error.jsx:13]: if `errorText` is required, why are we checking for its existence?

File: src/components/Error/Error.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/* eslint react/no-array-index-key: 0 */
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Alert, Button, Container } from 'react-bootstrap'
+
+const Error = ({ errorText, errorTitle, router, variant }) => {
+  return (
```
Comment (alishaevn) [inline src/components/Error/Error.jsx:7]: we can do an implicit return here.

File: src/components/Error/Error.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    <Container>
+      <Alert className='mt-5 text-break' variant={variant}>
+        {errorTitle && (
+          <Alert.Heading>{errorTitle}</Alert.Heading>
+        )}
+        {errorText && (errorText)}
+        <hr />
+        <div className='d-flex justify-content-end'>
+          <Button onClick={() => router.back()} variant={`outline-${variant}`}>
+            Click to return to the previous page.
```
Comment (alishaevn) [inline src/components/Error/Error.jsx:17]: an error redirects the user to a new page?

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: src/components/Error/Error.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/* eslint react/no-array-index-key: 0 */
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Alert, Button, Container } from 'react-bootstrap'
+
+const Error = ({ errors, showBackButton, router }) => {
+  const { errorTitle, errorText, variant} = errors
```
Comment (alishaevn) [inline src/components/Error/Error.jsx:7]:
```suggestion
  const { errorTitle, errorText, variant } = errors
```

**Review by alishaevn — CHANGES_REQUESTED** [review]

**Review by labradford — APPROVED** [review]

File: src/components/Error/Error.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/* eslint react/no-array-index-key: 0 */
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { Alert, Button, Container } from 'react-bootstrap'
+
+const Error = ({ errors, showBackButton, canDismissAlert, router }) => {
+  const { errorTitle, errorText, variant } = errors
+  const [show, setShow] = useState(true)
```
Comment (alishaevn) [inline src/components/Error/Error.jsx:8]:
```suggestion
  const [show, setShow] = useState(true)

```

**Review by alishaevn — CHANGES_REQUESTED** [review]

**Review by alishaevn — APPROVED** [review]

**Pull request merged**